midend/local_copyprop.cpp

```diff
@@ -322,6 +322,9 @@ IR::AssignmentStatement *DoLocalCopyPropagation::preorder(IR::AssignmentStatemen
 IR::AssignmentStatement *DoLocalCopyPropagation::postorder(IR::AssignmentStatement *as) {
     if (as->left->equiv(*as->right)) {
         LOG3("  removing noop assignment " << *as);
+        if (inferForFunc && inferForFunc->is_first_write_insert) {
```

Comment:
So why is this only done for the first write?
Can you also please add a comment to the is_first_write_insert field definition below documenting its meaning?

Reply:
Here the self-assignment is removed, but the expression on its left side still remains in the writes set, even though this assignment does nothing. So, we need to know whether that expression was in the set before this assignment (from some other assignments) or we are the first who put it there. If we are the first, then we need to remove it too.

I know this might not seem so efficient (removing a just inserted expression), but I suppose that in a 'reasonable' program the self-assignment should occur rarely (practically never). And constantly writing to the bool should be negligible overhead for this improvement.